src/components/buttons/primary-action-button/primary-action-button.spec.js

```diff
@@ -0,0 +1,184 @@
+
+const createProps = custom => ({
+  children: '',
+  ariaLabel: 'add product',
+  onClick: () => {},
+  ...custom,
+});
+
+describe('rendering', () => {
+  describe('PrimaryActionButton', () => {
+    describe('wrapper', () => {
+      let props;
+      let wrapper;
+      let buttonWrapper;
+      beforeEach(() => {
+        props = createProps({
+          isToggled: false,
+        });
+        wrapper = shallow(<PrimaryActionButton {...props} />);
+        buttonWrapper = wrapper.find(BaseButtonWrapper);
+      });
+      it('should render a <BaseButtonWrapper/>', () => {
+        expect(wrapper).toRender('BaseButtonWrapper');
+      });
+      it('should supply `id`', () => {
+        expect(buttonWrapper).toHaveProp('id', props.id);
+      });
+      it('should supply `name`', () => {
+        expect(buttonWrapper).toHaveProp('name', props.name);
+      });
+      it('should supply `ariaLabel`', () => {
+        expect(buttonWrapper).toHaveProp('ariaLabel', props.ariaLabel);
+      });
+      it('should supply `onClick`', () => {
+        expect(buttonWrapper).toHaveProp('onClick', props.onClick);
+      });
+
+      it('should supply `isToggled`', () => {
+        expect(buttonWrapper).toHaveProp('isToggled', props.isToggled);
+      });
+
+      describe('with data-* props', () => {
+        beforeEach(() => {
+          props = createProps({
+            dataAttr: {
+              'data-track-component': 'PrimaryActionButton',
+              'data-track-label': 'PrimaryActionButton',
+              'data-track-event': 'click',
+              'data-test': 'data-test-primary',
+            },
+          });
+          wrapper = shallow(<PrimaryActionButton {...props} />);
+          buttonWrapper = wrapper.find(BaseButtonWrapper);
+        });
+        it('should supply `dataAttr`', () => {
+          expect(buttonWrapper).toHaveProp(
+            'dataAttr',
+            expect.objectContaining({
+              'data-track-component': 'PrimaryActionButton',
+              'data-track-label': 'PrimaryActionButton',
+              'data-track-event': 'click',
+              'data-test': 'data-test-primary',
+            })
+          );
+        });
+      });
+
+      describe('setting the button as active', () => {
+        beforeEach(() => {
+          props = createProps({ isToggled: true });
+          wrapper = shallow(<PrimaryActionButton {...props} />);
+          buttonWrapper = wrapper.find(BaseButtonWrapper);
+        });
+
+        it('should pass the `isToggled` prop', () => {
+          expect(buttonWrapper).toHaveProp('isToggled', true);
+        });
+      });
+
+      describe('setting the button as disabled', () => {
+        beforeEach(() => {
+          props = createProps({ isDisabled: true });
+          wrapper = shallow(<PrimaryActionButton {...props} />);
+          buttonWrapper = wrapper.find(BaseButtonWrapper);
+        });
+
+        it('should pass the `isToggled` prop', () => {
```

Comment:
should pass the `isDisabled ` prop

Reply:
good catch 🦅👁

Reply:
its actually in the assertion below, not this one